irohad/consensus/yac/impl/timer_impl.cpp

```diff
@@ -21,17 +21,17 @@
 namespace iroha {
   namespace consensus {
     namespace yac {
-      void TimerImpl::invokeAfterDelay(uint64_t millis,
-                                       std::function<void()> handler) {
+      TimerImpl::TimerImpl(rxcpp::observable<Rep> invoke_delay)
+          : invoke_delay_(invoke_delay) {}
+
+      void TimerImpl::invokeAfterDelay(std::function<void()> handler) {
         deny();
-        handler_ = std::move(handler);
-        timer = rxcpp::observable<>::timer(std::chrono::milliseconds(millis));
-        handle = timer.subscribe_on(rxcpp::observe_on_new_thread())
-                     .subscribe([this](auto) { handler_(); });
+        handle_ = invoke_delay_.subscribe(
+            [handler{std::move(handler)}](auto) { handler(); });
```

Comment:
what is the point to move pointer? It is the same as a copy, isn't it?

Reply:
Function stores erased types (can be arbitrary big object), so It might be faster in some strange cases

Reply:
It might make sense if change handler's type to rvalue. In general why don't use perfect forwarding here?

Reply:
Update: mocking template function is a hassle, do we really need it?

Reply:
let it live like now, there's no real problem there